index.js

```diff
@@ -12,6 +12,15 @@ module.exports = function (obj) {
     (obj._isBuffer || // For Safari 5-7 (missing Object.prototype.constructor)
       (obj.constructor &&
       typeof obj.constructor.isBuffer === 'function' &&
-      obj.constructor.isBuffer(obj))
+      obj.constructor.isBuffer(obj)) ||
+      // Node v0.10
+      checkSlowBuffer(obj)
     ))
 }
+
+function checkSlowBuffer (obj) {
+  if (!obj.slice) return false
+  var buf = obj.slice(0, 0)
```

Comment:
Also, what if `slice` is a function that mutates some state in the program and the user wasn't expecting it to get called?

Also, what if `slice` throws an exception? This package should only return `true` or `false`, never throw.